Upstream-Status: Inappropriate

diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index a5a8d7d677..1f07b80945 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -488,7 +488,8 @@ check_val '' XINERAMA -lXinerama '' xinerama '' '' false
 check_lib '' XRANDR -lXrandr
 check_header '' XSHM X11/Xlib.h X11/extensions/XShm.h
 check_val '' XKBCOMMON -lxkbcommon '' xkbcommon 0.3.2 '' false
-check_val '' WAYLAND '-lwayland-egl -lwayland-client' '' wayland-egl 10.1.0 '' false
+check_lib '' WAYLAND '-lwayland-egl -lwayland-client'
+#check_val '' WAYLAND '-lwayland-egl -lwayland-client' '' wayland-egl 10.1.0 '' false
 check_val '' WAYLAND_CURSOR -lwayland-cursor '' wayland-cursor 1.12 '' false
 check_pkgconf WAYLAND_PROTOS wayland-protocols 1.15
 check_pkgconf WAYLAND_SCANNER wayland-scanner '1.15 1.12'
